src/registry.cr

```diff
@@ -0,0 +1,588 @@
+{% skip_file unless flag?(:win32) %}
+require "winerror"
+require "c/winreg"
+
+# This API povides access to the Windows registry. The main type is `Key`.
+#
+# ```
+# Registry::LOCAL_MACHINE.open("SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion", Registry::SAM::QUERY_VALUE) do |key|
+#   key.get_string("SystemRoot") # => "C:\\WINDOWS"
+# end
+# ```
+#
+# The Windows API defines some [predefined root keys](https://docs.microsoft.com/en-us/windows/desktop/sysinfo/predefined-keys) that are always open.
+# They are available as constants and can be used as entry points to the registry.
+#
+# * `HKEY_CLASSES_ROOT`
+# * `HKEY_CURRENT_USER`
+# * `HKEY_LOCAL_MACHINE`
+# * `HKEY_USERS`
+# * `HKEY_CURRENT_CONFIG`
+module Registry
+  # Information about file types and their properties.
+  CLASSES_ROOT = Key.new(LibC::HKEY_CLASSES_ROOT, "HKEY_CLASSES_ROOT")
```

Comment:
I'm curious why you removed the HKEY_ prefix on the const name?  As a user of this module I'd expect to find it with the HKEY_ prefix

Reply:
Because they're redundant. There are no namespaces in C, hence constants are often prefixed by a type identifier (like `HKEY`). In Crystal it's obvious that they are registry constants because they're defined in the `Registry` module.

Reply:
I can see your argument, but outside of the C-code context, the HKEY prefix has pervaded into everything.  It's even in the GUI regedit to the point that that is the actual name of the key now.  That was really the though process behind my statement of "As a user of this module I'd expect to find it with the HKEY_ prefix"